libusb/driver.c

```diff
@@ -0,0 +1,273 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libusb/driver.c
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Maciej Purski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <usbdriver.h>
+#include <sys/msg.h>
+#include <string.h>
+#include <unistd.h>
+
+
+static struct {
+	unsigned port;
+} usbdrv_common;
+
+
+int usb_connect(const usb_device_id_t *filters, int nfilters, unsigned drvport)
+{
+	msg_t msg = { 0 };
+	usb_msg_t *umsg = (usb_msg_t *)&msg.i.raw;
+	oid_t oid;
+
+	while (lookup("/dev/usb", NULL, &oid) < 0)
+		usleep(1000000);
+
+	msg.type = mtDevCtl;
+	msg.i.size = sizeof(usb_device_id_t) * nfilters;
+	msg.i.data = (void *)filters;
+
+	umsg->type = usb_msg_connect;
+	umsg->connect.port = drvport;
+	umsg->connect.nfilters = nfilters;
+
+	if (msgSend(oid.port, &msg) < 0)
+		return -1;
+
+	usbdrv_common.port = oid.port;
+
+	return oid.port;
+}
+
+
+int usb_eventsWait(int port, msg_t *msg)
+{
+	unsigned long rid;
+
+	if (msgRecv(port, msg, &rid) < 0)
```

Comment:
In theory msgRecv can be interrupted by signal and return -EINTR. Maybe (or something along these lines):
```suggestion
	int err;
	do {
		err = msgRecv(port, msg, &rid);
		if (err < 0 && err != -EINTR)
			return -1;
	} while (err == -EINTR);
```

Reply:
Looks good, I'll change it.